app/models/piece.rb

```diff
@@ -32,4 +32,64 @@ def capture!(piece_at_location)
     update_attributes(x_position: nil, y_position: nil)
   end
 
+  def space_occupied?(x, y)
+    game.pieces.where('x_space = ? AND y_space = ?', x, y).present?
   end
+
+  def vertical_move?(x, y)
+    return true if x_space == x && y_space != y
```

Comment:
I don't understand the syntax here. Is this a Ruby thing? All of my if statements are set up like "if condition" "return boolean" not the opposite.

Reply:
Ruby is super lenient.